tests/integration/protected_product.t

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+
+use File::Basename "dirname";
+
+use Storable qw(dclone);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_application_ready();
+
+my $ua = new_client();
+
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+create_user($ua, \%create_user_args);
+
+# Create some products
+
+my %product_form = (
+	generic_name => "A generic name",
+	ingredients_text => "apple, milk, eggs, palm oil",
+	categories => "some unknown category",
+	labels => "organic",
+	origin => "france",
+	brands => "Test brand",
+	packaging_text_en =>
+		"1 wooden box to recycle, 6 25cl glass bottles to reuse, 3 steel lids to recycle, 1 plastic film to discard",
+	'nutriment_energy-kj' => 10,
+	nutriment_fat => 5,
+	nutriment_proteins => 2,
+	quantity => "90g",
+);
+
+my @products = (
+	{
+		%{dclone(\%default_product_form)},
+		%{dclone(\%product_form)},
+		(
+			code => '0200000000034',
+			product_name => "An unprotected product",
+		)
+	},
+	{
+		%{dclone(\%default_product_form)},
+		%{dclone(\%product_form)},
+		(
+			code => '0200000000035',
+			product_name => "A protected product",
+		)
+	},
+	{
+		%{dclone(\%default_product_form)},
+		%{dclone(\%product_form)},
+		(
+			code => '0200000000134',
+			product_name => "An unprotected product",
+		)
+	},
+	{
+		%{dclone(\%default_product_form)},
+		%{dclone(\%product_form)},
+		(
+			code => '0200000000135',
+			product_name => "A protected product",
+		)
+	},
+);
+
+foreach my $product_form_override (@products) {
+	edit_product($ua, $product_form_override);
+}
+
+# Declare which fields have been sent by an owner and should be protected
+for my $code ('0200000000035', '0200000000135') {
+	my $product_ref = retrieve_product($code);
+	$product_ref->{owner_fields} = {
+		"categories" => 1680183938,
+		"labels" => 1680183938,
+		"brands" => 1680183938,
+		"product_name_en" => 1680183938,
+		"generic_name_en" => 1680183938,
+		"ingredients_text_en" => 1680183938,
+		"energy-kj" => 1680183938,
+		"fat" => 1680183938,
+		"proteins" => 1680183938,
+		"quantity" => 1680183938,
+	};
+	store_product("organization-owner", $product_ref, "protecting the product");
+}
+
+# Note: expected results are stored in json files, see execute_api_tests
+my $tests_ref = [
```

Comment:
If I understand it well, you "get-edited-protected-product-web-form" test_case depends upon "edit-protected-product-web-form" (it's a verification)

I think it would be better to pair them in the tests_ref structure and then have, but it would not enable using execute_api_tests.

So we shall at least comment on that !

```suggestion
# Note: expected results are stored in json files, see execute_api_tests
# Each test is composed of two test case: 
# one that edits a product, 
# and one that get the product to verify if it was edited or proctected
my $tests_ref = [
```

Reply:
Sounds good.